packages/legend-query/src/stores/QueryBuilderExplorerState.ts

```diff
@@ -310,14 +396,36 @@ export const getQueryBuilderPropertyNodeData = (
     mappingNodeData,
   );
   if (propertyNode.type instanceof Class) {
-    propertyNode.childrenIds = propertyNode.type
-      .getAllProperties()
-      .concat(propertyNode.type.getAllDerivedProperties())
-      .map((p) => `${propertyNode.id}.${p.name}`);
+    propertyNode.childrenIds = generateChildrenIds(propertyNode);
   }
   return propertyNode;
 };
 
+export const getQueryBuilderSubTypeNodeData = (
+  subClass: Class,
+  parentNode: QueryBuilderExplorerTreeNodeData,
+): QueryBuilderExplorerTreeSubTypeNodeData => {
+  const subTypeNode = new QueryBuilderExplorerTreeSubTypeNodeData(
+    `${
+      parentNode instanceof QueryBuilderExplorerTreeRootNodeData
+        ? `@${subClass.name}`
+        : `${parentNode.id}->subType(@${subClass.path})`
+    }`,
+    `@${subClass.name}`,
+    `${
+      parentNode instanceof QueryBuilderExplorerTreeRootNodeData
+        ? `$x->subType(@${subClass.path})`
+        : `${parentNode.dndText}->subType(@${subClass.path})`
+    }`,
+    subClass,
+    parentNode.id,
+    false,
+    { mapped: true, skipMappingCheck: true }, //display subclasses, anyway. To do: Enchance mapping algo to take into account this
```

Comment:
Please move the commnent to above this line, also please correct `To do` to `TODO` as that's our standard keyword. Also, @MauricioUyaguari do you think we want to do the algo for this? We can right?

Reply:
@YannanGao-gs  Could you investigate the logic needed for this and update the issue for the subtype. We need to figure out what to check in the mapping to figure out what properties in the subtype are mapped.